api/kv/value/value.go

```diff
@@ -33,19 +34,23 @@ type Value struct {
 	vtype    Type
 	numeric  uint64
 	stringly string
+
+	// Collections
+	collection interface{}
 	// TODO Lazy value type?
 }
 
 const (
-	INVALID Type = iota // No value.
-	BOOL                // Boolean value, use AsBool() to get it.
-	INT32               // 32 bit signed integral value, use AsInt32() to get it.
-	INT64               // 64 bit signed integral value, use AsInt64() to get it.
-	UINT32              // 32 bit unsigned integral value, use AsUint32() to get it.
-	UINT64              // 64 bit unsigned integral value, use AsUint64() to get it.
-	FLOAT32             // 32 bit floating point value, use AsFloat32() to get it.
-	FLOAT64             // 64 bit floating point value, use AsFloat64() to get it.
-	STRING              // String value, use AsString() to get it.
+	INVALID    Type = iota // No value.
+	BOOL                   // Boolean value, use AsBool() to get it.
+	INT32                  // 32 bit signed integral value, use AsInt32() to get it.
+	INT64                  // 64 bit signed integral value, use AsInt64() to get it.
+	UINT32                 // 32 bit unsigned integral value, use AsUint32() to get it.
+	UINT64                 // 64 bit unsigned integral value, use AsUint64() to get it.
+	FLOAT32                // 32 bit floating point value, use AsFloat32() to get it.
+	FLOAT64                // 64 bit floating point value, use AsFloat64() to get it.
+	STRING                 // String value, use AsString() to get it.
+	COLLECTION             // Array value, use AsCollection()
```

Comment:
I'm concerned that one day we'll agree to add dictionary values to the specification, at which point the use of "Collection" will be ambiguous. Can we use ARRAY?

Reply:
Sure.